src/entities/activity/lib/services/AnswersUploadService.ts

```diff
@@ -399,6 +399,10 @@ export class AnswersUploadService implements IAnswersUploadService {
 
     const identifier = data.userIdentifier && encrypt(data.userIdentifier);
 
+    const eventHistoryId =
+      data.eventVersion && data.eventId
```

Comment:
Nit: `data.eventId` should always be truthy since it's not nullable and would have to be an invalid UUID provided by the BE to become falsy, so I'd omit this bit (aligns with the web app):
```suggestion
      data.eventVersion
```